### DISCUSSION TITLE
Unable to get authentication

### DISCUSSION THREAD

**Discussion by atikjaz** [spencerwooo/onedrive-vercel-index#658] [Q&A]
I'm using E5 developer stuck at last step to get refresh tokens.

**Comment by myl7** [comment]
Could you provide more detailed log? like console log, error messages or screenshots.

**Reply by atikjaz** [reply]
> Could you provide more detailed log? like console log, error messages or screenshots.

![Screenshot_2022-05-24-12-26-39-39_40deb401b9ffe8e1df2f1cc5ba480b12](https://user-images.githubusercontent.com/64772699/169968041-25d5cd4f-9f23-48a4-ba1a-c651b3310450.jpg)
This is what I get.

**Reply by myl7** [reply]
You might try to set `userPrincipleName` field of config file `config/site.config.js` to your account name (which is also your account email) to resolve it.

**Reply by atikjaz** [reply]
> You might try to set `userPrincipleName` field of config file `config/site.config.js` to your account name (which is also your account email) to resolve it.

I did it ends with` @w35k3.onmicrosoft.com`

**Reply by osh-2022-ta** [reply]
A full email address is required other than only the domain part. It should be like `xxx@w35k3.onmicrosoft.com`. If you do not want to public your email, you can also pass it via Vercel envs.

**Reply by osh-2022-ta** [reply]
Sry for replying with another wrong account in case you are confused

**Reply by atikjaz** [reply]
> A full email address is required other than only the domain part. It should be like `xxx@w35k3.onmicrosoft.com`. If you do not want to public your email, you can also pass it via Vercel envs.

It is full email address I was just making short here.

**Reply by myl7** [reply]
Well the error message is only generated when two `userPrincipleName` are different, which are compared here:

https://github.com/spencerwooo/onedrive-vercel-index/blob/152380bd057d256bcd6cf3ececefcf5826fecaf9/pages/onedrive-vercel-index-oauth/step-3.tsx#L58

So maybe you can add a `console.log(...)` there to see what the 2 `userPrincipleName` are actually. The output will be in the Vercel runtime log.

**Reply by atikjaz** [reply]
> Well the error message is only generated when two `userPrincipleName` are different, which are compared here:
> 
> https://github.com/spencerwooo/onedrive-vercel-index/blob/152380bd057d256bcd6cf3ececefcf5826fecaf9/pages/onedrive-vercel-index-oauth/step-3.tsx#L58
> 
> So maybe you can add a `console.log(...)` there to see what the 2 `userPrincipleName` are actually. The output will be in the Vercel runtime log.

I redeploy it again from the scratch and now I'm stuck on step 2 to get localhost link I'm being redirected to `https://login.partner.microsoftonline.cn/.....` for verification and it has different login details.
